Initial blog PR: "Using heuristic evaluation to inform the user exper… (#3091)
opensearch-project/project-website: _posts/2024-07-19-using-heuristic-evaluation-to-inform-the-user-experience-strategy-of-a-product.markdown

To improve the UI / UX of the OpenSearch Playground, one of the primary ways that new users can evaluate the OpenSearch platform, we tried to view our product through the fresh eyes of a new customer. To do so we deployed the research principles of a UX heuristics evaluation. In this process, we identified four key recommendations. These recommendations are informed by the most common violations of tenets and traps and UX heuristics that our researchers identified in the Playground and are summarized below.

![](/assets/media/blog-images/2024-07-19-using-heuristic-evaluation-to-inform-the-user-experience-strategy-of-a-product/Image.jpg)
**Focus on workflows that help customers achieve a goal rather than complete specific tasks**. The Playground home page offers customers concrete tasks like ingesting data, exploring data, and trying a query assistant. Think about organizing the site to support customer goals such as proactive anomaly monitoring, visualizing your data, and getting started with minimal expertise.

> **Use consistent terminology and workflows across all tools and features to help customers work faster and build intuitive mental models.** There are several instances where the experience changes depending on whether users are logged in or not. There are also instances where terms like “buckets” are used that differ from industry standards. Simplifying terminology and workflows to follow industry knowledge and best practices will increase customer confidence that this product is for them.

Reviewer: ```suggestion
1. **Use consistent terminology and workflows across all tools and features to help customers work faster and build intuitive mental models.** There are several instances where the experience changes depending on whether users are logged in or not. There are also instances where terms are used that differ from industry standards, like "buckets." Simplifying terminology and workflows to follow industry best practices will increase customer confidence in the product.
```

> **Minimize available actions within the Playground that trial users cannot access**. A recurring Tenet & Trap violation was Inviting Dead End. Users encounter options (e.g. “add sample data”) within the Playground that they do not have access to and are unable to perform. Focusing on eliminating these encounters and errors make it easier for potential customers to assess the platform.

Reviewer: ```suggestion
1. **Minimize available actions in OpenSearch Playground that trial users cannot access**. A recurring Tenet & Trap violation was Inviting Dead End. Users encounter options (for example, "add sample data") that they do not have access to and are unable to perform. Focusing on eliminating these encounters and errors makes it easier for potential customers to assess the product.
```

> **Investigate potential content gaps on [OpenSearch.org](http://opensearch.org/) which** may hinder new user trial. The customer journey on [OpenSearch.org](http://opensearch.org/) seems to have information gaps that add friction to understanding and trialing of the Playground.

Reviewer: ```suggestion
1. **Investigate potential content gaps on [OpenSearch.org](http://opensearch.org/)** that may hinder a new user trial. The website appears to have information gaps that add friction to understanding and trialing OpenSearch Playground.
```

## Takeaways

> * Perform a heuristics analysis to rapidly identify common traps and violations in your product design

Reviewer: ```suggestion
* Perform a heuristic analysis to rapidly identify common traps and violations in your product design.
```

> * You can scope your research as broadly or narrowly as you need

Reviewer: ```suggestion
* You can scope your research as broadly or narrowly as needed.
```

> * UX heuristics insights are often universally applicable to your entire UX; be sure to broaden your understanding (and dissemination) of your findings to ensure maximum value from your efforts

Reviewer: ```suggestion
* UX heuristic insights are often universally applicable to your entire UX; be sure to broaden your understanding (and dissemination) of your findings to ensure maximum derived value from your efforts.
```

> * You can engage outside UX research experts or utilize existing resources to apply these UI Tenets and Traps and 10 Usability Heuristics to your product

Reviewer: ```suggestion
* You can engage outside UX research experts or utilize existing resources to apply these UI Tenets & Traps and Nielsen's 10 Usability Heuristics to your product.
```

> * Implement a regular Heuristics research practice to monitor and track improvements.

Reviewer: ```suggestion
* Implement a regular heuristic research practice to monitor and track improvements.
```

> ## References

Reviewer: ```suggestion
```

> ## References

Reviewer: Deleted this section, as we already link to and explain these resources earlier in the blog post.

> [UI Tenets & Traps](https://uitraps.com/)

Reviewer: ```suggestion
```

> [Nielsen’s 10 Usability Heuristics](https://www.nngroup.com/articles/ten-usability-heuristics/)

Reviewer: ```suggestion
```

### About Steyer

"We don't just analyze data, we create meaning from it. We don't just report insights, we inspire action from them."

> Steyer partners with our clients to make meaningful connections between people and information, solving our clients’ business problems with the tools we know best: strategic analysis, user research, and the creation, organization, and revision of business content. With nearly three decades of delighting both our clients and our own team members, we understand what it takes to effect real change in the real world.

Reviewer: ```suggestion
Steyer partners with our clients to make meaningful connections between people and information, solving our clients' business problems with the tools we know best: strategic analysis, user research, and the creation, organization, and revision of business content. With nearly three decades of delighting both our clients and our own team members, we understand what it takes to effect real change in the real world.
```

![](/assets/media/blog-images/2024-07-19-using-heuristic-evaluation-to-inform-the-user-experience-strategy-of-a-product/Image1.jpg)

> **About Ray Salas - Steyer Insight Research Lead and Strategist**

Reviewer: ```suggestion
```

> **About Ray Salas - Steyer Insight Research Lead and Strategist**

Reviewer: Deleted, as this information is already provided in rsalas.md. If the image is not already included in that file, please relocate.

> As a leader with 20+ years of team building and people management experience at companies like JP Morgan Chase, Microsoft, and Amazon, Ray has been both a creator and consumer of customer research. As a research leader, he has implemented frameworks that deeply integrate human insight into engineering and product lifecycles, and he has driven prioritizations and roadmaps by balancing human insight, tech capability, and business viability. Above all, Ray believes in product and service development that is customer led and not just customer validated.

Reviewer: ```suggestion
```

> ![](/assets/media/blog-images/2024-07-19-using-heuristic-evaluation-to-inform-the-user-experience-strategy-of-a-product/Image2.jpg)

Reviewer: ```suggestion
```
